src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -62,29 +62,31 @@ protected void starting(Description description) {
     }
 
     private void startRecording(Description des) {
-        GraphicsConfiguration gc = GraphicsEnvironment
-                .getLocalGraphicsEnvironment()
-                .getDefaultScreenDevice()
-                .getDefaultConfiguration();
-
-        String mimeType = FormatKeys.MIME_QUICKTIME;
-        String videoFormatName = VideoFormatKeys.ENCODING_QUICKTIME_ANIMATION;
-        String compressorName = VideoFormatKeys.COMPRESSOR_NAME_QUICKTIME_ANIMATION;
-        Dimension outputDimension = gc.getBounds().getSize();
-        int bitDepth = BIT_DEPTH;
-        float quality = QUALITY_RATIO;
-        int screenRate = FRAME_RATE_PER_SEC;
-        long maxRecordingTime = MAX_RECORDING_TIME_SECS;
-
-        Format outputFormatForScreenCapture = getOutputFormatForScreenCapture(videoFormatName,
-                compressorName, outputDimension,
-                bitDepth, quality, screenRate);
-
         try {
+            GraphicsConfiguration gc = GraphicsEnvironment
+                    .getLocalGraphicsEnvironment()
+                    .getDefaultScreenDevice()
+                    .getDefaultConfiguration();
+
+            String mimeType = FormatKeys.MIME_QUICKTIME;
+            String videoFormatName = VideoFormatKeys.ENCODING_QUICKTIME_ANIMATION;
+            String compressorName = VideoFormatKeys.COMPRESSOR_NAME_QUICKTIME_ANIMATION;
+            Dimension outputDimension = gc.getBounds().getSize();
+            int bitDepth = BIT_DEPTH;
+            float quality = QUALITY_RATIO;
+            int screenRate = FRAME_RATE_PER_SEC;
+
+            Format outputFormatForScreenCapture = getOutputFormatForScreenCapture(videoFormatName,
+                    compressorName, outputDimension,
+                    bitDepth, quality, screenRate);
+
             this.screenRecorder = new JUnitScreenRecorder
                     (gc, gc.getBounds(), getFileFormat(mimeType),
                             outputFormatForScreenCapture, null, null, diagnostics);
             this.screenRecorder.start();
+        } catch (UnsupportedOperationException e) {
+            // E.g. java.awt.HeadlessException
+            logger.warn("Exception starting test recording {}", e);
```

Comment:
Or simply

```java
} catch (Exception e) {
    logger.warn("Exception starting test recording {}", e);
}
```

If you have some kind of visceral dislike of catching `Exception` (personally I have never understood what the problem is—we know that _any_ kind of exception ought to be caught here, and reported) you could use

```java
} catch (HeadlessException | IOException | AWTException e) {
    logger.warn("Exception starting test recording {}", e);
}
```